src/osd/scrubber/pg_scrubber.h

```diff
@@ -660,6 +685,16 @@ class PgScrubber : public ScrubPgIF, public ScrubMachineListener {
 
 private:
 
+  /**
+   *  preventing WaitDigestUpdate state from reacting to DigestUpdate events
+   *  once scrub_finish() is called.
+   *  Note - making this a separate FSM state would have opened the same race window
+   *  we are trying to minimize. There is no safe way to have the non-reentrant Boost
+   *  state-chart transition into a new state without going through the queue (which
+   *  also means releasing the PG lock).
+   */
+  bool m_finish_sequence_started{false};
```

Comment:
I don't think I understand how we can get a stray DigestUpdate.

Reply:
removed all that 'finishing flag' stuff. I made a minor change to the logic that sens the event, and hope it really won't happen.